tests/test_nestedattr.py

```diff
@@ -0,0 +1,44 @@
+from datetime import date
+from pprint import pprint
+
+from marshmallow import Schema, fields
+
+from marshmallow_utils.fields import nestedattr
+
+
+class ArtistSchema(Schema):
+    """Artist Schema."""
+
+    name = fields.Str()
+
+
+class AlbumSchema(Schema):
+    """Album Schema."""
+
+    artist = nestedattr.NestedAttribute(ArtistSchema())
+
+
+class MyAlbum(dict):
+
+    @property
+    def artist(self):
+        return {'name': 'property'}
+
+
+album = MyAlbum(
+    artist={'name': 'david'},  # this shouldn't be used by the schema
+    title="Hunky Dory",
+    release_date=date(1971, 12, 17)
+)
+
+
+def test_test_schema():
+    album = MyAlbum(
+        title="Hunky Dory",
+        release_date=date(1971, 12, 17)
```

Comment:
doubt: below it dumps this object which *doesn't have* any `artist`... How does it actually work then? Maybe I'm not understanding the usage of this field. @slint Can confirm.

Reply:
I suggest adding something like this also:

```python
album = MyAlbum(
    artist={'name': 'david'},  # this shouldn't be used in the schema dump
    title="Hunky Dory",
    release_date=date(1971, 12, 17)
)
```

Maybe then the `artist` property could be something like:

```python
class MyAlbum(dict):

    @property
    def artist(self):
        return {'name': self['artist']['name'].catpialize()}
```

to showcase that we prefer the property/attribute's behavior compared to `__getitem__` from the dict